src/ts/settlement.ts

```diff
@@ -132,25 +132,82 @@ export type EncodedSettlement = [
   [Interaction[], Interaction[], Interaction[]],
 ];
 
+/**
+ * An object listing all flag options in order along with their bit offset.
+ */
+export const flagMasks = {
+  kind: {
+    offset: 0,
+    options: [OrderKind.SELL, OrderKind.BUY],
+  },
+  partiallyFillable: {
+    offset: 1,
+    options: [false, true],
+  },
+  signingScheme: {
+    offset: 2,
+    options: [
+      SigningScheme.EIP712,
+      SigningScheme.ETHSIGN,
+      SigningScheme.EIP1271,
+      SigningScheme.PRESIGN,
+    ],
+  },
+} as const;
+
+function encodeFlag<
+  K extends keyof typeof flagMasks,
+  V extends typeof flagMasks[K]["options"][number]
+>(key: K, flag: V): number {
+  const index = flagMasks[key].options.findIndex(
+    (search: unknown) => search === flag,
+  );
+  if (index === undefined) {
+    throw new Error(`Bad key/value pair to encode: ${key}/${flag}`);
```

Comment:
If the caller respects the types, yes!
I wrote it like this because the TS compiler complained that `index` could be undefined and because it would be a weird behavior to debug if called incorrectly from pure JS (`undefined  << 1 === 0`)

Reply:
It makes a lot of sense to have this case (like you said, we build an NPM package that exposes JS code - we do similar things when in other `switch (enum) { .. }` code where, in theory, the `default` should never be needed). It was more me making sure I understood everything :smile:.